osquery/events/linux/auditdnetlink.cpp

```diff
@@ -325,11 +326,25 @@ bool AuditdNetlinkReader::configureAuditService() noexcept {
 
   // Rules required by the process_events table
   if (FLAGS_audit_allow_process_events) {
-    VLOG(1) << "Enabling audit rules for the process_events table";
+    VLOG(1) << "Enabling audit rules for the process_events (execve, execveat) "
+               "table";
 
-    for (int syscall : kProcessEventsSyscalls) {
+    for (int syscall : kExecProcessEventsSyscalls) {
       monitored_syscall_list_.insert(syscall);
     }
+
+    if (FLAGS_audit_allow_fork_process_events) {
+      VLOG(1) << "Enabling audit rules for the process_events (fork, vfork, "
+                 "clone) table";
+
+      for (int syscall : kForkProcessEventsSyscalls) {
+        monitored_syscall_list_.insert(syscall);
+      }
+    }
+
+  } else if (FLAGS_audit_allow_fork_process_events) {
+    LOG(WARNING) << "--audit_allow_fork_process_events is ignored without "
```

Comment:
This can be compared to enabling `--audit_allow_config` without also having the command line flag that enables Audit.

If we want to conform to the rest of the code, we can just remove the warning log. Otherwise, we could implement a flag validator and make sure that both flags are active.

Which one would you prefer?

Reply:
Ah sorry, my comment was too general and not helpful. You are right in that we should address/explore gflags more completely later.

Always your call!